### DISCUSSION TITLE
Multiple OnPublish Commands

### DISCUSSION THREAD

**Discussion by t3therdev** [bluenviron/mediamtx#285] [Questions]
## Which version are you using?

v0.13.0 OS WINDOWS

## Question
Multiple OnPublish Commands

We are currently using onpublish to record streams to a file using ffmpeg. At the same time we would like to create an HLS stream. How can we create multiple onpublish actions?

Also, when using ffmpeg and onpublish, is there a way to log the output? it seems we only see it on the console but can't log it?

**Comment by t3therdev** [comment]
Ok seems we can just add additional onpublish lines. So it is working.

**Comment by t3therdev** [comment]
working.

**Comment by t3therdev** [comment]
actually it seems only the last one added is working. if we have to onpublish commands.

**Comment by dorinclisu** [comment]
By utilizing multiple `runOnPublish` you are only replacing the previous value, which is why only the last one is running (it's just the way YAML works).

I think it's possible to achieve what you want by simply utilizing the parallel process shell feature. Running `CMD1 & CMD2` will run CMD1 in background and wait for CMD2 to finish. This works in linux shells and might work on widnows shell. Now there might be trouble sending interrupt signal to CMD1 once the publishing is done or the server quits, which is why you might want to wrap the commans in a shell file and instead call  `runOnPublish: ./cmd.sh`. Inside the cmd file you would implement the signal handling for the spawned processes.

**Comment by aler9** [comment]
You can do exactly what @dorinclisu said - i'm noticing that there are a lot of Windows users, and the Windows problem is that the shell engine is really old and weak, exactly the opposite of the Linux one - but at the moment i don't know what to do. I could

* incorporate a scripting engine into the server - the server would become huge and there'll be license problems
* make the configuration more powerful, the problem is that nothing is as versatile as a scripting engine

**Comment by t3therdev** [comment]
We are going to build an external application that will be called by the onpublish command, which is working fine, when a stream publishing our app triggers. The issue is that we also want a trigger when the publisher disconnects. Is it possible to achieve this?

[a comment by an automated account is not shown]
